### PR TITLE
Enh/env vars

### LINKED ISSUE #109: Set environment variables from `sim.make.exec`
If I understand well, overriding environment variables requires modifying the Snakefile:

https://snek5000.readthedocs.io/en/latest/configuring.html#overriding-configuration-with-environment-variable

It would be nice to be able to override environment variables from `sim.make.exec`.

### PR DESCRIPTION
**Pull request by paugier**
Fixes #109  (and #101).

It should be possible to modify Snake configuration without modifying the Snakefile.

### CODE REVIEW

**Comment by paugier** [comment]
`pyyaml` is already used by Snake so actually it does not add any dependency.

**Comment by ashwinvis** [comment]
This PR introduces two new environment variables which can affect the compilation - which I would really like to avoid.

- `SNEK_CONFIGFILE`: Instead of this, how about we decide to always write `config_simul.yml` during `sim.output.post_init()` via a method `sim.output.write_snakemake_config(custom_env_vars={...})`. The method can be again invoked in the user script.
- `SNEK_ENV_SENSITIVE`: Confusing and unnecessary. Set it as either True or False in the user `Snakefile`.

**Comment by paugier** [comment]
> SNEK_ENV_SENSITIVE: Confusing and unnecessary. Instead set either True or False in the user Snakefile while calling update_snakemake_config(env_sensitive=...)

I don't understand. I'm not going to modify the Snakefile to run one particular script launching a simulation.

I understand that the default behavior should be to not take into account environment variables (so in the Snakefile, we should not use `env_sensitive=True`), but it has to be easy to change this behavior.

**Comment by ashwinvis** [comment]
Setting an environment variable which overrides if the application is sensitive to environment variables or not .... is kind of a circular logic :smile: . If you strongly feel that the `env_sensitive is None` behaviour is useful, fine by me, but keep the default value `env_sensitive=False`.

**Comment by paugier** [comment]
> but keep the default value env_sensitive=False.

I think it does not make sense. It should be possible to get the very standard behavior in Unix and Windows of sensibility to environment variables without modifying the Snakefile. One should be able to do that with solvers written following `snek5000-canonical`.

Users should not have to modify the Snakefile. I think we have to remember that for most users and writers of snek solvers, the Snakemake file is an internal detail of the Snek5000's implementation.

I'm not strongly attached to the exact name `SNEK_ENV_SENSITIVE` so we can find something more explicit.

Regarding the other environment variable `SNEK_CONFIGFILE`, this is more an internal detail of my implementation of `exec(..., env_vars_configfile={...})`. If we can find a better solution to implement this feature, I don't care about `SNEK_CONFIGFILE`.

**Comment by paugier** [comment]
> SNEK_CONFIGFILE: Instead of this, how about we decide to always write config_simul.yml during sim.output.post_init() via a method sim.output.write_snakemake_config(custom_env_vars={...}). The method can be again invoked in the user script.

I'm going to study this idea.

**Comment by paugier** [comment]
Indeed, no need for `SNEK_CONFIGFILE`. It's now removed and the config file is always written in the simulation directory.

We could even depreciate `get_configfile` and just write in the Snakemake file

```
configfile: ./config_simul.yml
```

File: src/snek5000/output/base.py [hunk 1]
```diff
@@ -294,6 +301,12 @@ def update_snakemake_config(
 
             append_debug_flags(config, warnings)
 
+            if env_sensitive is None:
+                env_sensitive = os.environ.get("SNEK_UPDATE_CONFIG_ENV_SENSITIVE", False)
+                if isinstance(env_sensitive, str):
+                    # correct for "0", "false", "False"
+                    env_sensitive = bool(yaml.safe_load(env_sensitive))
```
Comment (ashwinvis) [inline src/snek5000/output/base.py:308]: Could use `distutils.util.strtobool` here.

**Comment by ashwinvis** [comment]
This is a much cleaner approach. Thanks!

File: src/snek5000/output/base.py (hunk 1 above)
Comment (paugier) [reply src/snek5000/output/base.py:308]: I thought about it but now that `distutils` is deprecated, I'd prefer to avoid it.

Comment (ashwinvis) [reply src/snek5000/output/base.py:324]: Good point.

File: src/snek5000/output/base.py [hunk 2]
```diff
@@ -578,6 +607,35 @@ def write_makefile_usr(self, template, fp=None, **template_vars):
...
+
+        custom_env_vars: dict (None)
+            Environment variables used to update the configuration found by
+            :meth:`find_configfile`.
+
+        """
+        if mpi.rank != 0:
+            return
+
+        path_configfile = self.find_configfile()
```
Comment (ashwinvis) [inline src/snek5000/output/base.py:624]: We can pass a `host` parameter to `find_configfile` here.

Comment (paugier) [reply src/snek5000/output/base.py]: Done in d434825f170b939e91f337d2b88c6fc4219676e3

**Pull request merged**